L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc

```diff
@@ -0,0 +1,339 @@
+      for (int etaIndex = -etaHalfSize; etaIndex <= etaHalfSize; etaIndex++) {
+        for (int phiIndex = -phiHalfSize; phiIndex <= phiHalfSize; phiIndex++) {
+          if ((etaIndex == 0) && (phiIndex == 0))
+            continue;
+          if (phiIndex > 0) {
+            if (phiIndex > -etaIndex) {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt > getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            } else {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt >= getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            }
+          } else {
+            if (phiIndex >= -etaIndex) {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt > getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            } else {
+              isLocalMaximum =
+                  ((isLocalMaximum) && (centralPt >= getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex)));
+            }
+          }
+        }
+      }
+      if (isLocalMaximum) {
+        seeds.emplace_back(std::make_tuple(iEta, iPhi));
+      }
+    }
+  }
+
+  return seeds;
+}
+
+reco::CaloJet Phase1L1TJetProducer::buildJetFromSeed(const TH2F& caloGrid, const std::tuple<int, int>& seed) {
+  int iEta = std::get<0>(seed);
+  int iPhi = std::get<1>(seed);
+
+  int etaHalfSize = (int)jetIEtaSize_ / 2;
+  int phiHalfSize = (int)jetIPhiSize_ / 2;
+
+  float ptSum = 0;
+  // Scanning through the grid centered on the seed
+  for (int etaIndex = -etaHalfSize; etaIndex <= etaHalfSize; etaIndex++) {
+    for (int phiIndex = -phiHalfSize; phiIndex <= phiHalfSize; phiIndex++) {
+      ptSum += getTowerEnergy(caloGrid, iEta + etaIndex, iPhi + phiIndex);
+    }
+  }
+
+  // Creating a jet with eta phi centered on the seed and momentum equal to the sum of the pt of the components
+  reco::Candidate::PolarLorentzVector ptVector;
+  ptVector.SetPt(ptSum);
+  ptVector.SetEta(caloGrid.GetXaxis()->GetBinCenter(iEta));
+  ptVector.SetPhi(caloGrid.GetYaxis()->GetBinCenter(iPhi));
+  reco::CaloJet jet;
+  jet.setP4(ptVector);
+  return jet;
+}
+
+std::vector<reco::CaloJet> Phase1L1TJetProducer::_buildJetsFromSeedsWithPUSubtraction(
+    const TH2F& caloGrid, const std::vector<std::tuple<int, int>>& seeds, bool killZeroPt) {
+  // For each seed take a grid centered on the seed of the size specified by the user
+  // Sum the pf in the grid, that will be the pt of the l1t jet. Eta and phi of the jet is taken from the seed.
+  std::vector<reco::CaloJet> jets;
+  for (const auto& seed : seeds) {
+    reco::CaloJet jet = buildJetFromSeed(caloGrid, seed);
+    subtract9x9Pileup(caloGrid, jet);
+    //killing jets with 0 pt
+    if ((vetoZeroPt_) && (jet.pt() <= 0))
+      continue;
+    jets.push_back(jet);
+  }
+  return jets;
+}
+
+std::vector<reco::CaloJet> Phase1L1TJetProducer::_buildJetsFromSeeds(const TH2F& caloGrid,
```

Comment:
function can be `const` etc.

Reply:
done and removed `caloGrid` input: https://github.com/cms-l1t-offline/cmssw/blob/fc3a1a454f40efdf7c95dd97295a6a085a674092/L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc#L306-L307